sympy/solvers/ode.py

```diff
@@ -2279,7 +2280,7 @@ def _expand(expr):
     # things like -C1, so rerun constantsimp() one last time before returning.
     for i, eqi in enumerate(eq):
         eq[i] = constantsimp(eqi, constants)
-        eq[i] = constant_renumber(eq[i], 'C', 1, 2*order)
+        eq[i] = constant_renumber(eq[i], 'C', 1, 2*order,dont_change_constants = eq_constants)
```

Comment:
`exclude` is the term we usually use in this context (instead of `dont_change_constants`)